UndertaleModLib/Decompiler/Decompiler.cs

```diff
@@ -4013,7 +4013,7 @@ public static void BuildSubFunctionCache(UndertaleData data)
                             {
                                 lock (data.KnownSubFunctions)
                                 {
-                                    data.KnownSubFunctions.Add(assign.Destination.Var.Name.Content, funcDef.Function);
+                                    data.KnownSubFunctions.TryAdd(assign.Destination.Var.Name.Content, funcDef.Function);
```

Comment:
instead of blindly ignoring duplicate entries, is there a reason why it ever tries to add duplicate entries in the first place?

Reply:
I am not sure for now, but I feel that the `foreach` loop above for `data.Functions` is redundant.

Reply:
See #1586.

Reply:
can you slap a TODO onto it then please?